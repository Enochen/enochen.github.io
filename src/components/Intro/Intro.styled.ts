import styled, { css } from 'styled-components'
import { rhythm, scale, TABLET_MEDIA_QUERY } from '../../utils/typography'
import { Section } from '../Section'

export const Header = styled.div`
  text-align: center;
  display: flex;
  flex-direction: column;
  padding-top: 25vh;
  transition: padding-top 0.5s;
`

export const BaseNameGroup = styled.div`
  margin: auto;
`

export const NameGroup = styled(BaseNameGroup)`
  &:hover {
    color: #6460b8;
    cursor: pointer;
  }
`

const LinkStyle = css`
  color: inherit;
  transition: font-size 0.5s;
`

export const BaseName = styled.h1`
  ${scale(2.3)};
`

export const Name = styled(BaseName)`
  ${LinkStyle}
`

export const BaseDesc = styled.h2`
  ${scale(1)};
`

export const Desc = styled(BaseDesc)`
  ${LinkStyle}
`

export const About = styled.div`
  ${scale(0.2)};
  visibility: hidden;
  opacity: 0;
  text-align: justify;
  width: 80%;
  margin: 0 auto;
  padding-top: ${rhythm(1.7)};
  max-height: 0;
  overflow: hidden;
  transition: height 0.5s;
`

export const IconGroup = styled.div`
  display: flex;
  justify-content: space-between;
  text-align: center;
  margin-top: auto;
  padding-top: 0;
  padding-bottom: 10vh;
`

export const IntroIcon = styled.div`
  margin: auto;
  &:hover {
    color: #6460b8;
    cursor: pointer;
  }
`

export const IconLink = styled.a`
  color: hsla(0, 0%, 0%, 0.7);
`

export const IconLabel = styled.div`
  user-select: none;
`

interface SectionProps {
  readonly aboutActive?: boolean
}

export const Intro = styled(Section)<SectionProps>`
  padding-bottom: 0;
  max-width: 45rem;

  ${props =>
    props.aboutActive &&
    css`
      ${Header} {
        padding-top: 15vh;
      }
      ${Name} {
        ${scale(1.8)};
        padding-bottom: 0rem;
      }
      ${Desc} {
        ${scale(0.6)};
      }
      ${About} {
        visibility: visible;
        opacity: 1;
        max-height: 1000px;
        ${scale(0.2)};
        transition: font-size 0.5s, max-height 0.5s, visibility 0.5s,
          opacity 0.5s;
      }
    `}

  ${TABLET_MEDIA_QUERY} {
    .icon {
      display: none;
    }
    ${Name} {
      ${scale(1.8)};
    }
    ${Desc} {
      ${scale(0.6)};
    }
    ${About} {
      padding-top: ${rhythm(1)};
    }
    ${IconGroup} {
      flex-wrap: wrap;
      padding-bottom: 13vh;
    }
    ${IntroIcon} {
      display: flex;
      justify-content: center;
      width: 30%;
      padding: 0.5rem 2rem;
    }
    ${IconLabel} {
      text-transform: uppercase;
      font-weight: 600;
      padding: 0.5rem 2rem;
    }
    ${props =>
      props.aboutActive &&
      css`
        ${Name} {
          font-size: 3rem;
        }
        ${Desc} {
          font-size: 1.2rem;
        }
        ${About} {
          ${scale(0.2)};
        }
      `}
  }
`
