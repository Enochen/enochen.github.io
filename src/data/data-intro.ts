import { IconType } from 'react-icons'
import {
  FaFlask,
  FaGithub,
  FaGraduationCap,
  FaLinkedin,
  FaNewspaper,
  FaRegUser,
  FaUser,
} from 'react-icons/fa'

export interface IIcon {
  icon: IconType
  iconAlt?: IconType
  label: string
  link?: string
  external?: boolean
}

export interface IIntro {
  name: string
  desc?: string
  aboutData?: string[]
  iconData?: IIcon[]
}

export const IntroData: IIntro = {
  name: 'Enoch Chen',
  desc: 'cs @ cornell',
  aboutData: [
    'I am currently an undergrad at Cornell University studying Computer Science. I am passionate about designing effective and reliable systems. My goal is to make things that others can enjoy well.',
    'In my free time, I like tinkering with keyboards.',
    'Check out some of my stuff <a href = "#projects">here</a>.',
  ],
  iconData: [
    {
      icon: FaUser,
      iconAlt: FaRegUser,
      label: 'About',
    },
    {
      icon: FaFlask,
      label: 'Projects',
      link: '#projects',
    },
    {
      icon: FaGithub,
      label: 'Github',
      link: 'http://github.com/Enochen',
      external: true,
    },
    {
      icon: FaLinkedin,
      label: 'LinkedIn',
      link: 'https://www.linkedin.com/in/enochen728/',
      external: true,
    },
    {
      icon: FaGraduationCap,
      label: 'Transcript',
      link: '/misc/2020-01-04-Transcript.pdf',
      external: true,
    },
    {
      icon: FaNewspaper,
      label: 'Resume',
      link: '/misc/Enoch Chen resume.pdf',
      external: true,
    },
  ],
}
