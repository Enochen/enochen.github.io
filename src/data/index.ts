export * from './data-intro'
export * from './data-project'