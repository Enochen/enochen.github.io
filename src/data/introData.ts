import { IconType } from 'react-icons'
import {
  FaFlask,
  FaGithub,
  FaGraduationCap,
  FaLinkedin,
  FaNewspaper,
  FaRegUser,
  FaUser,
} from 'react-icons/fa'

interface IIcon {
  icon: IconType
  icon2?: IconType
  label: string
  link?: string
  external?: boolean
}

interface IIntro {
  name: string
  desc: string
  aboutData: string[]
  iconData: IIcon[]
}

export const IntroData: IIntro = {
  name: 'Enoch Chen',
  desc: 'cs @ cornell',
  aboutData: [
    'I am currently an undergrad at Cornell University studying Computer Science. I am passionate about designing effective and reliable systems. My goal is to make things that others can enjoy well.',
    'In my free time, I like tinkering with keyboards.',
    'Check out some of my stuff <a href = "#projects">here</a>.',
  ],
  iconData: [
    {
      icon: FaUser,
      icon2: FaRegUser,
      label: 'About',
    },
    {
      icon: FaFlask,
      label: 'Projects',
      link: '#projects',
    },
    {
      icon: FaGithub,
      label: 'Github',
      link: 'http://github.com/Enochen',
      external: true,
    },
    {
      icon: FaLinkedin,
      label: 'LinkedIn',
      link: 'https://www.linkedin.com/in/enochen728/',
      external: true,
    },
    {
      icon: FaGraduationCap,
      label: 'Transcript',
      link: '/misc/2020-01-04-Transcript.pdf',
      external: true,
    },
    {
      icon: FaNewspaper,
      label: 'Resume',
      link: '/misc/Enoch Chen resume.pdf',
      external: true,
    },
  ],
}
