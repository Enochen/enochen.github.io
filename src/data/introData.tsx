import React from 'react'
import { OutboundLink } from 'gatsby-plugin-gtag'
import { IconType } from 'react-icons'

import {
  FaFlask,
  FaGithub,
  FaGraduationCap,
  FaLinkedin,
  FaNewspaper,
  FaRegUser,
  FaUser,
} from 'react-icons/fa'

export interface IconInfo {
  icon: IconType
  iconAlt?: IconType
  label: string
  link?: string
  external?: boolean
}

export interface IntroInfo {
  name: string
  desc?: string
  aboutData?: JSX.Element[]
  iconData?: IconInfo[]
}

export const IntroData: IntroInfo = {
  name: 'Enoch Chen',
  desc: 'cs @ cornell',
  aboutData: [
    <>
      I am currently an undergrad at Cornell University studying Computer
      Science. I am passionate about designing effective and reliable systems,
      and I love getting the details just right. I want to make things that
      others can enjoy!
    </>,
    <>
      In my free time, you can find me juggling, tinkering with keyboards, or
      just thinking about life.
    </>,
    <>
      Check out some of my stuff{' '}
      <OutboundLink href="#projects">here</OutboundLink>.
    </>,
  ],
  iconData: [
    {
      icon: FaRegUser,
      iconAlt: FaUser,
      label: 'About',
    },
    {
      icon: FaFlask,
      label: 'Projects',
      link: '#projects',
    },
    {
      icon: FaGithub,
      label: 'Github',
      link: 'http://github.com/Enochen',
      external: true,
    },
    {
      icon: FaLinkedin,
      label: 'LinkedIn',
      link: 'https://www.linkedin.com/in/enochen728/',
      external: true,
    },
    {
      icon: FaNewspaper,
      label: 'Resume',
      link: '/misc/Enoch Chen resume.pdf',
      external: true,
    },
  ],
}
