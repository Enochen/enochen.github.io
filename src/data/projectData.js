export default [
  {
    name: "cue",
    desc:
      "I joined <a href='https://cornelldti.org' target='_blank' rel='noopener'>Cornell Design & Tech Initiative</a> in Fall 2019 as a Full Stack Developer. My team is responsible for cue, an event-discovery app for Cornell students at Ithaca. I developed a set of scripts to facilitate populating our event database with external data.",
    tags: ["HTML/CSS", "Javascript", "Python", "ReactJS", "Django"],
    live: "https://www.cornelldti.org/Projects/events",
    github: "https://github.com/cornell-dti/events-backend/",
    image: "cue.jpg",
  },
  {
    name: "Ask Palette",
    desc:
      "Generate color palettes from your search terms! Uses Google Cloud Vision to extract colors from search terms. Developed as a hackathon project at <a href='https://bigredhacks.com' target='_blank' rel='noopener'>BigRed//Hacks</a>.",
    tags: ["HTML/CSS", "Python", "Flask", "Google Cloud"],
    live: "https://askpalette.appspot.com/",
    github: "https://github.com/Enochen/askpalette",
    image: "askpalette.png",
  },
  {
    name: "Half Bot Half Brain",
    desc:
      "It's like Mad Libs, but better. The backend generates grammatically correct text from input with the help of spaCy. Developed as a hackathon project at <a href='https://teenhacksli.com/' target='_blank' rel='noopener'>TeenhacksLI</a>.",
    tags: ["HTML/CSS", "JavaScript", "Python", "spaCy"],
    live: "https://halfbothalfbrain.netlify.com/",
    github: "https://github.com/ageor0114/nyit-hackathon",
    image: "hbhb.png",
  },
]