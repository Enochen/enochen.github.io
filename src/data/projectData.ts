export interface ProjectInfo {
  name: string
  desc: string
  tags: string[]
  live?: string
  github?: string
  image: string
}

export interface ProjectList {
  data: ProjectInfo[]
}

export const ProjectData: ProjectList = {
  data: [
    {
      name: 'cue',
      desc:
        'I joined <a href="https://cornelldti.org" target="_blank" rel="noopener">Cornell Design & Tech Initiative</a> in Fall 2019 as a Full Stack Developer. My team is responsible for cue, an event-discovery app for Cornell students at Ithaca. I developed a set of scripts to facilitate populating our event database with external data.',
      tags: ['HTML/CSS', 'Javascript', 'Python', 'ReactJS', 'Django'],
      live: 'https://www.cornelldti.org/Projects/events',
      github: 'https://github.com/cornell-dti/events-backend/',
      image: 'cue.jpg',
    },
    {
      name: 'Ask Palette',
      desc:
        'Generate color palettes from your search terms! Uses Google Cloud Vision to extract colors from search terms. Developed as a hackathon project at <a href="https://bigredhacks.com" target="_blank" rel="noopener">BigRed//Hacks</a>.',
      tags: ['HTML/CSS', 'Python', 'Flask', 'Google Cloud'],
      live: 'https://askpalette.appspot.com/',
      github: 'https://github.com/Enochen/askpalette',
      image: 'askpalette.png',
    },
    {
      name: 'This Website',
      desc:
        'Over winter break, I decided to learn some new technologies and use them to put something together. It seemed like a good idea to make a personal website, and this is what came out of that idea.',
      tags: ['HTML/CSS', 'TypeScript', 'GatsbyJS', 'GraphQL', 'React/Preact' ],
      github: 'https://github.com/Enochen/portfolio',
      image: 'portfolio.png',
    },
    {
      name: 'Half Bot Half Brain',
      desc:
        'It\'s like Mad Libs, but better. The backend generates grammatically correct text from input with the help of spaCy. Developed as a hackathon project at <a href="https://teenhacksli.com/" target="_blank" rel="noopener">TeenhacksLI</a>.',
      tags: ['HTML/CSS', 'JavaScript', 'Python', 'spaCy'],
      live: 'https://halfbothalfbrain.netlify.com/',
      github: 'https://github.com/ageor0114/nyit-hackathon',
      image: 'hbhb.png',
    },
  ],
}
