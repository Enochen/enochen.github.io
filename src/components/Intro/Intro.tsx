import React, { FC, useState } from 'react'
import { IconContext } from 'react-icons'
import { goToTop, removeHash } from 'react-scrollable-anchor'
import { IconInfo, IntroInfo } from '../../data/introData'
import * as styled from './Intro.styled'

interface IconProps extends IconInfo {
  action?: () => void
}

const Icon: FC<IconInfo> = ({
  icon,
  label,
  link,
  external = false,
  action,
}) => {
  const IconElement = icon
  const OptionalLink = link ? styled.IconLink : React.Fragment

  const linkHref = link
    ? {
        href: link,
      }
    : {}
  const linkProps = external
    ? {
        target: '_blank',
        rel: 'noopener noreferrer',
      }
    : {}
  const wrapperProps = action
    ? {
        onClick: action,
      }
    : {}

  return (
    <styled.IntroIcon {...wrapperProps}>
      <OptionalLink {...linkHref} {...linkProps}>
        <IconElement />
        <styled.IconLabel>{label}</styled.IconLabel>
      </OptionalLink>
    </styled.IntroIcon>
  )
}

const ActionIcon: FC<IconProps> = ({
  icon,
  label,
  action,
}) => {
  const IconElement = icon
  const wrapperProps = action
    ? {
        onClick: action,
      }
    : {}

  return (
    <styled.IntroIcon {...wrapperProps}>
        <IconElement />
        <styled.IconLabel>{label}</styled.IconLabel>
    </styled.IntroIcon>
  )
}

const makeAbout = (data: string[]) => {
  return data.map((x, i) => (
    <p key={i} dangerouslySetInnerHTML={{ __html: x }} />
  ))
}

const makeIcons = (data: IconInfo[], action: () => void, altIcon: boolean) => {
  return (
    <IconContext.Provider value={{ size: '42', className: 'icon' }}>
      {data.map((x, i) =>
        x.link ? (
          <Icon key={i} {...x}></Icon>
        ) : (
          <ActionIcon
            key={i}
            {...x}
            action={action}
            icon={altIcon ? x.iconAlt : x.icon}
          ></ActionIcon>
        )
      )}
    </IconContext.Provider>
  )
}

export const Intro: FC<IntroInfo> = ({
  name,
  desc,
  aboutData = [],
  iconData = [],
}) => {
  const [aboutOn, setAboutOn] = useState(false)
  const toggleAboutOn = () => {
    setAboutOn(!aboutOn)
    goToTop()
    removeHash()
  }
  const about = makeAbout(aboutData)
  const icons = makeIcons(iconData, toggleAboutOn, aboutOn)
  return (
    <styled.Intro aboutActive={aboutOn}>
      <styled.Header>
        <styled.NameGroup onClick={toggleAboutOn}>
          <styled.Name>{name}</styled.Name>
          <styled.Desc>{desc}</styled.Desc>
        </styled.NameGroup>
        <styled.About>{about}</styled.About>
      </styled.Header>
      <styled.IconGroup>{icons}</styled.IconGroup>
    </styled.Intro>
  )
}
