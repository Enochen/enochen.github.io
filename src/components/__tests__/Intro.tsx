import React from 'react'
import { FaFlask, FaRegUser, FaUser } from 'react-icons/fa'
import renderer from 'react-test-renderer'
import { IntroInfo } from '../../data'
import { Intro } from '../Intro/Intro'
describe('Intro Section', () => {
  it('renders correctly', () => {
    const introData: IntroInfo = {
      name: 'George Washington',
      desc: 'SWE @ Apple',
      aboutData: [
        <p>I am the first President of the United States.</p>,
        <p>I am very honest. I robbed an Apple store when I was 6 years old. When my dad confronted me about it, I told the truth. That's how I became the President of the United States.</p>,
        <p>My website can be found <a href = "https://www.georgie.com">here</a>.</p>,
      ],
      iconData: [
        {
          icon: FaUser,
          iconAlt: FaRegUser,
          label: 'About Me!',
        },
        {
          icon: FaFlask,
          label: 'My Science Experiments',
          link: '#science',
        },
      ],
    }
    const tree = renderer.create(<Intro {...introData} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
