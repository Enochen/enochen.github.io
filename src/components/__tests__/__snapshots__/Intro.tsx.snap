// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro Section renders correctly 1`] = `
<div
  className="sc-AxjAm kiCCiV sc-fznZeY dxuBCw"
>
  <div
    className="sc-AxhCb cPqAfH"
  >
    <div
      className="sc-AxhUy sc-AxgMl cpKqWr"
      onClick={[Function]}
    >
      <h1
        className="sc-AxheI sc-Axmtr hlBzfA"
      >
        George Washington
      </h1>
      <h2
        className="sc-AxmLO sc-fzozJi YwgyU"
      >
        SWE @ Apple
      </h2>
    </div>
    <div
      className="sc-fzoLsD bcHiPe"
    >
      <p
        dangerouslySetInnerHTML={
          Object {
            "__html": "I am the first President of the United States.",
          }
        }
      />
      <p
        dangerouslySetInnerHTML={
          Object {
            "__html": "I am very honest. I robbed an Apple store when I was 6 years old. When my dad confronted me about it, I told the truth. That's how I became the President of the United States.",
          }
        }
      />
      <p
        dangerouslySetInnerHTML={
          Object {
            "__html": "My website can be found <a href = \\"https://www.georgie.com\\">here</a>.",
          }
        }
      />
    </div>
  </div>
  <div
    className="sc-fzpans gAmEVq"
  >
    <div
      className="sc-fzplWN kNoqvi"
      onClick={[Function]}
    >
      <svg
        className="icon"
        fill="currentColor"
        height="42"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width="42"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
        />
      </svg>
      <div
        className="sc-fznKkj fsBdSp"
      >
        About Me!
      </div>
    </div>
    <div
      className="sc-fzplWN kNoqvi"
    >
      <a
        className="sc-fznyAO jolFEl"
        href="#science"
      >
        <svg
          className="icon"
          fill="currentColor"
          height="42"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="42"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M437.2 403.5L320 215V64h8c13.3 0 24-10.7 24-24V24c0-13.3-10.7-24-24-24H120c-13.3 0-24 10.7-24 24v16c0 13.3 10.7 24 24 24h8v151L10.8 403.5C-18.5 450.6 15.3 512 70.9 512h306.2c55.7 0 89.4-61.5 60.1-108.5zM137.9 320l48.2-77.6c3.7-5.2 5.8-11.6 5.8-18.4V64h64v160c0 6.9 2.2 13.2 5.8 18.4l48.2 77.6h-172z"
          />
        </svg>
        <div
          className="sc-fznKkj fsBdSp"
        >
          My Science Experiments
        </div>
      </a>
    </div>
  </div>
</div>
`;
